File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            
            if token.lstrip('-').isnumeric():
                stack.append(int(token))
            else:
                num2 = stack.pop();
                num1 = stack.pop();
                if token == "+":
                    stack.append(num1 + num2)
                elif token == "-":
                    stack.append(num1 - num2)
                elif token == "*":
                    stack.append(num1 * num2)
                elif token == "/":
                    stack.append(int(num1 / num2))
        return stack[-1]

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stack = Vec::new();
        
        for token in tokens {
            if let Ok(num) = token.parse::<i32>(){
                stack.push(num);
            } 
            else {
                if let (Some(num2), Some(num1)) = (stack.pop(), stack.pop()){
                    match token.as_str() {
                       "+" => { stack.push(num1 + num2) },
                       "-" => { stack.push(num1 - num2)  },
                       "*" => { stack.push(num1 * num2)  },
                       "/" => { stack.push(num1 / num2)  },
                        _ => (),
                    }
                }
        
                else {
                    return 0;
                }
                
               
            }
        }
        *stack.last().unwrap()
    }
}